app.js

```diff
@@ -1,17 +1,137 @@
 import express from 'express';
-import db from './db/db';
+import teaStorage from './db/teaStorage';
 import bodyParser from 'body-parser';
 
 const uuidv1 = require('uuid/v1');
-
 const app = express();
+const URL = require("url").URL;
+const httpResponse = require('./response/response');
+
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({extended: false}));
 
 
+app.get('/teatime/tea/all', (request, response) => {
+    return httpResponse.successWithResponse(response, teaStorage, 'teas retrieved successfully');
+});
+
+function teaCheckAreRequiredParamsValid(reqBody, response) {
+    if (!reqBody.name) {
+        httpResponse.badRequestOnMissingParam(response, 'name');
+        return false;
+    } else if (!reqBody.description) {
+        httpResponse.badRequestOnMissingParam(response, 'description');
+        return false;
+    } else if (!reqBody.originCountry) {
+        httpResponse.badRequestOnMissingParam(response, 'originCountry');
+        return false;
+    } else if (!reqBody.harvestSeason) {
+        httpResponse.badRequestOnMissingParam(response, 'harvestSeason');
+        return false;
+    } else if (!reqBody.caffeineContent) {
+        httpResponse.badRequestOnMissingParam(response, 'caffeineContent');
+        return false;
+    }
+    return true;
+}
+
+//----tea controllers ----
+const stringIsAValidUrl = (s) => {
+    try {
+        new URL(s);
+        return true;
+    } catch (error) {
+        return false;
+    }
+};
```

Comment:
🗡 **[Logic]:** To Ci raczej moja droga nie zadziała :)
![Screenshot 2019-12-02 at 19 02 30](https://user-images.githubusercontent.com/40358447/69983198-667b5f00-1536-11ea-8e06-3aa3656049f1.jpg)

Zobacz wstawiłem 2 adresy, jeden dobry, drugi zły -> w obu wypadkach mam `true` 😅

Reply:
XDDD ale na stack overflow mówili, że działa. kłamcy